volumes:
  prometheus_data:
    external: true

services:
  prometheus:
    image: docker.io/prom/prometheus:latest
    container_name: prometheus
    ports:
      - "9090:9090"
    volumes:
      - ./prometheus.yaml:/etc/prometheus/prometheus.yml:Z
      - "prometheus_data:/prometheus"
    restart: unless-stopped
    command:
      - "--config.file=/etc/prometheus/prometheus.yml"
    labels:
      io.containers.autoupdate: "registry"

  node_exporter:
    image: quay.io/prometheus/node-exporter:latest
    container_name: node_exporter
    command:
      - "--path.rootfs=/host"
    pid: host
    restart: unless-stopped
    volumes:
      - "/:/host:ro,rslave"
    labels:
      io.containers.autoupdate: "registry"

  # Changed `/var/run/fail2ban/` permissions `sudo chown -R $(whoami):$(whoami) /var/run/fail2ban` and
  # added `pid: host` to the service definition to allow the container to access the host's fail2ban socket
  # TODO: Find a way to run this container as a non-root user
  fail2ban:
    image: registry.gitlab.com/hctrdev/fail2ban-prometheus-exporter
    container_name: fail2ban-exporter
    volumes:
      - /var/run/fail2ban/:/var/run/fail2ban:Z
    ports:
      - "9191:9191"
    pid: host
    labels:
      io.containers.autoupdate: "registry"
